sherpa/optmethods/optfcts.py

```diff
@@ -89,58 +91,56 @@ def _narrow_limits( myrange, xxx, debug ):
+                print('x = ', my_x, ' is < lower limit = ', my_l)
             if my_x > my_h:
-                print 'x = ', my_x, ' is > upper limit = ', my_h
-
-    def raise_min_limit( range, xmin, x, debug=False ):
-        myxmin = numpy.asarray( map( lambda xx: xx - range * numpy.abs(xx),
-                                     x ), numpy.float_ )
-        if False != debug:
-            print
-            print 'raise_min_limit: myxmin=%s' % myxmin    
-            print 'raise_min_limit: x=%s' % x
+                print('x = ', my_x, ' is > upper limit = ', my_h)
+
+    def raise_min_limit(range, xmin, x, debug=False):
+        myxmin = numpy.asarray(list(map(lambda xx: xx - range * numpy.abs(xx), x)), numpy.float_)
+        if debug:
+            print()
+            print('raise_min_limit: myxmin=%s' % myxmin)    
+            print('raise_min_limit: x=%s' % x)
         below = numpy.flatnonzero(myxmin < xmin)
         if below.size > 0:
             myxmin[below] = xmin[below]
-        if False != debug:
-            print 'raise_min_limit: myxmin=%s' % myxmin    
-            print 'raise_min_limit: x=%s' % x
-            print
+        if debug:
+            print('raise_min_limit: myxmin=%s' % myxmin)    
+            print('raise_min_limit: x=%s' % x)
+            print()
         return myxmin
 
-    def lower_max_limit( range, x, xmax, debug=False ):
-        myxmax = numpy.asarray( map( lambda xx: xx + range * numpy.abs(xx),
-                                     x ), numpy.float_ )
-        if False != debug:
-            print
-            print 'lower_max_limit: x=%s' % x
-            print 'lower_max_limit: myxmax=%s' % myxmax    
+    def lower_max_limit(range, x, xmax, debug=False):
+        myxmax = numpy.asarray(list(map(lambda xx: xx + range * numpy.abs(xx), x)), numpy.float_)
+        if debug:
+            print()
+            print('lower_max_limit: x=%s' % x)
+            print('lower_max_limit: myxmax=%s' % myxmax)    
         above = numpy.flatnonzero(myxmax > xmax)
         if above.size > 0:
             myxmax[above] = xmax[above]
-        if False != debug:
-            print 'lower_max_limit: x=%s' % x
-            print 'lower_max_limit: myxmax=%s' % myxmax    
-            print
+        if debug:
+            print('lower_max_limit: x=%s' % x)
+            print('lower_max_limit: myxmax=%s' % myxmax)    
+            print()
         return myxmax
 
     x = xxx[ 0 ]
     xmin = xxx[ 1 ]
     xmax = xxx[ 2 ]
 
     if False != debug:
-        print 'narrow_limits: xmin=%s' % xmin
-        print 'narrow_limits: x=%s' % x        
-        print 'narrow_limits: xmax=%s' % xmax
+        print('narrow_limits: xmin=%s' % xmin)
+        print('narrow_limits: x=%s' % x)        
+        print('narrow_limits: xmax=%s' % xmax)
     myxmin = raise_min_limit( myrange, xmin, x, debug=False )
     myxmax = lower_max_limit( myrange, x, xmax, debug=False )
 
     if False != debug:
```

Comment:
I guess you don't want to do cleanup like change this to

    if debug:

I guess this might not be quite the same, since debug need not be a boolean, but earlier checks in this function rely on `if debug` so it should be safe here.